download-data.py

```diff
@@ -155,151 +477,124 @@ def main() -> None:
-        patterns += [f"*_{level}_*.h5ad" for level in includes]
-        patterns += [f"*_{level}_*.hdf5" for level in includes]
-
-    # If projects or samples are specified, extend the file-only patterns to specify parent directories
-    if args.projects:
-        patterns = add_parent_dirs(patterns, args.projects.split(","))
-    elif args.samples:
-        patterns = add_parent_dirs(patterns, args.samples.split(","))
-
-    # if samples are specified, bulk is excluded, as it is not associated with individual samples
-    if include_bulk and not args.samples:
-        if args.projects:
-            patterns += [
-                f"{project}/*_bulk_*.tsv" for project in args.projects.split(",")
-            ]
-        else:
-            patterns += ["*_bulk_*.tsv"]
-
-    ### Add patterns to the sync command and run it! ###
-    for p in patterns:
-        sync_cmd += ["--include", p]
-
-    if args.dryrun:
-        sync_cmd += ["--dryrun"]
-
-    if args.profile:
-        sync_cmd += ["--profile", args.profile]
-
-    if args.test_data:
-        sync_cmd += ["--no-sign-request"]
-
-    subprocess.run(sync_cmd, check=True)
+    # list results modules and exit if that was what was requested
+    if args.list_results or args.module_results:
+        all_modules = get_results_modules(
+            bucket=results_bucket, release=release, profile=args.profile
+        )
+    if args.list_results:
+        print(
+            f"Available module results for release {release}:\n",
+            "\n".join(all_modules),
+            sep="\n",
+        )
+        return
 
-    ### Print summary messages ###
-    print("\n\n\033[1mDownload Summary\033[0m")  # bold
-    print("Release:", release)
-    print("Data Format:", ", ".join(formats))
-    print("Processing levels:", ", ".join(includes))
-    print("Downloaded data to:", args.data_dir / release)
+    # check that the requested modules are available
+    if args.module_results:
+        modules = set(args.module_results.split(","))
+        if not modules.issubset(all_modules):
+            print(
+                f"One or more requested modules are not available for release {release}.",
+                "Available modules are:\n",
+                "\n".join(all_modules),
+                sep="\n",
+                file=sys.stderr,
+            )
+            sys.exit(1)
 
-    ### Update current link to point to new or test data ###
-    # only do this if we are using test data or the specified release is "current" or "latest", not for specific dates
-    if (
-        args.test_data or args.release.lower() in ["current", "latest"]
-    ) and not args.dryrun:
-        # update the current symlink
-        current_symlink = args.data_dir / "current"
-        current_symlink.unlink(missing_ok=True)
-        current_symlink.symlink_to(release)
-        print(f"Updated 'current' symlink to point to '{release}'.")
+    ### Download the data ###
+    if not args.module_results:
```

Comment:
Looking at this I really do want a flag that specifies download type... I think the default could be release data, but I like being explicit about what type of data you are trying to get. Some people could think that by having a `--module_results` argument, you are adding download of results to download of a release, but these are separate. 

Another kind of wild thought, but what if we had two scripts... We can pull out the functions you have into a separate `utils` script that gets read in to each of the scripts. Then you have one that's just for downloading results from the workflow and one just for downloading release data.